### PR TITLE
More passes cleanups

### PR DESCRIPTION
**Pull request by donggyukim**
* ONLY NEEDS LGTM
* SHOULD BE MERGED BY #267

### CODE REVIEW

File: src/main/scala/firrtl/passes/PadWidths.scala [hunk 1]
```diff
@@ -3,77 +3,56 @@ package passes
...
+      Mux(cond, fixup(width(tpe))(tval), fixup(width(tpe))(fval), tpe)
+    case e: ValidIf => e copy (value = fixup(width(e.tpe))(e.value))
+    case e: DoPrim => e.op match {
+      case Lt | Leq | Gt | Geq | Eq | Neq | Not | And | Or | Xor |
+           Add | Sub | Mul | Div | Rem | Shr =>
+        // sensitive ops
+        e map fixup((e.args map (width(_)) foldLeft 0)(math.max(_, _)))
+      case Dshl =>
+        // special case as args aren't all same width
+        e copy (args = Seq(fixup(width(e.tpe))(e.args(0)), e.args(1)))
```
Comment (azidar) [inline src/main/scala/firrtl/passes/PadWidths.scala]: Dshl -> Dshlw

File: src/main/scala/firrtl/passes/PadWidths.scala [hunk 2]
```diff
@@ -3,77 +3,56 @@ package passes
...
+      case Shl =>
+        // special case as arg should be same width as result
+        e copy (args = Seq(fixup(width(e.tpe))(e.args(0))))
```
Comment (azidar) [inline src/main/scala/firrtl/passes/PadWidths.scala]: Shl -> Shlw

File: src/main/scala/firrtl/passes/Passes.scala [hunk 3]
```diff
@@ -249,78 +231,48 @@ object Legalize extends Pass {
...
+    if (v_keywords(n)) "%s$".format(n) else n
+  def verilog_rename_e(e: Expression): Expression = e match {
+    case e: WRef => e copy (name = verilog_rename_n(e.name))
+    case e => e map verilog_rename_e
+  }
+  def verilog_rename_s(s: Statement): Statement =
+    s map verilog_rename_s map verilog_rename_e map verilog_rename_n
+
+  def run(c:Circuit): Circuit =
+    c copy (modules = (c.modules map (_ map verilog_rename_s)))
```
Comment (azidar) [inline src/main/scala/firrtl/passes/Passes.scala]: Need to map on ports as well

File: src/main/scala/firrtl/passes/LowerTypes.scala [hunk 4]
```diff
@@ -81,165 +78,145 @@ object LowerTypes extends Pass {
...
+          case 1 => exps.head
+          case _ => error("Error! lowerTypesExp called on MemKind " + 
+                          "SubField that needs to be expanded!")(info, mname)
         }
-        case e: Mux => e map (lowerTypesExp)
-        case e: ValidIf => e map (lowerTypesExp)
-        case e: DoPrim => e map (lowerTypesExp)
-        case e @ (_: UIntLiteral | _: SIntLiteral) => e
-      }
+      case _ => WRef(loweredName(e), e.tpe, kind(e), gender(e))
```
Comment (azidar) [inline src/main/scala/firrtl/passes/LowerTypes.scala]: Rewrite with copy

File: src/main/scala/firrtl/passes/LowerTypes.scala [hunk 5]
```diff
@@ -81,165 +78,145 @@ object LowerTypes extends Pass {
...
+    }
+    case e: Mux => e map lowerTypesExp(memDataTypeMap, info, mname)
+    case e: ValidIf => e map lowerTypesExp(memDataTypeMap, info, mname)
+    case e: DoPrim => e map lowerTypesExp(memDataTypeMap, info, mname)
+    case _ => e
```
Comment (azidar) [inline src/main/scala/firrtl/passes/LowerTypes.scala]: Don't do this, match on WRef specifically, and for the case _ => ..., throw an error. We want to error the compiler if we see a WSubAccess

**Comment by azidar** [comment]
last four commits lgtm once changes are done

File: src/main/scala/firrtl/passes/LowerTypes.scala (hunk 4 above)
Comment (donggyukim) [inline src/main/scala/firrtl/passes/LowerTypes.scala]: They are different types?

File: src/main/scala/firrtl/passes/PadWidths.scala (hunk 1 above)
Comment (donggyukim) [inline src/main/scala/firrtl/passes/PadWidths.scala]: This is a good point!

**Comment by jackkoenig** [comment]
Confirmed no functional changes across the 4 commits

**Pull request merged**